UNRELEASED.md

```diff
@@ -12,6 +12,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Bug fixes
 
+- Fixed vertical misalignment in `Banner.Header`([#870](https://github.com/Shopify/polaris-react/pull/870))
```

Comment:
^ @dpersing added! Let me know if I should update the format (first time contributing to Polaris)